packages/shared/profiles/reducer.ts

```diff
@@ -41,6 +46,51 @@ export function profileReducer(state?: ProfileState, action?: AnyAction): Profil
           }
         }
       }
+    case PROFILES_REQUEST:
+      return {
+        ...state,
+        userInfo: {
+          ...state.userInfo,
+          [action.payload.userId]: { ...state.userInfo[action.payload.userId], status: 'loading' }
+        }
+      }
+    case PROFILES_SUCCESS:
+      const { profiles } = (action as ProfilesSuccessAction).payload
+      const profilesState = profiles.reduce((newState, profile) => {
+        return {
+          ...newState,
+          [profile.userId]: {
+            ...state.userInfo[profile.userId],
+            data: profile,
+            status: 'ok'
+          }
+        }
+      }, {})
+
+      return {
+        ...state,
+        userInfo: {
+          ...state.userInfo,
+          ...profilesState
+        }
+      }
+
+    case PROFILES_FAILURE:
+      const { userIds } = (action as ProfilesFailureAction).payload
+      const newProfilesState = userIds.reduce((newState, userId) => {
+        return {
+          ...newState,
+          [userId]: { status: 'error', data: action.payload.error }
+        }
+      }, {})
```

Comment:
Kernel code prefers vanilla loops when possible, mainly for performance and secondly for readability. Please apply to all occurrences of .reduce

```suggestion
      const newProfilesState: Record<string, ProfileUserInfo> = {}
      for (const userId of userIds) {
        newProfilesState[userId] = { status: 'error', data: action.payload.error }
      }
```

Reply:
Okay, will replace the reduces